rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -445,8 +445,7 @@ def parse_value_string(type_, value_string):
     if type_.is_primitive_type() and not type_.is_array:
         return parse_primitive_value_string(type_, value_string)
 
-    # TODO(mikaelarguedas) change this condition once string escape function is implemented
-    if type_.is_primitive_type() and type_.is_array and type_.type != 'string':
+    if type_.is_primitive_type() and type_.is_array:
```

Comment:
Do we need to make it optional? It sounds easier to just require quoting and escaping of interior quotes. It also makes it easier for a human (or syntax highlighter) to determine what is a literal and what is syntax. I'd even say, require double quote for the outside quotes. This isn't Python, and just as many languages, if not more, would use single quotes only for single characters.